deployment/cloud/modules/server/main.tf

```diff
@@ -0,0 +1,221 @@
+            "ec2:DisassociateAddress"
+          ],
+          Resource = "*",
+          Condition = {
+            StringEquals = {
+              "aws:ResourceTag/Project" = data.aws_default_tags.default_tags.tags.Project
+            }
+          }
+        }
+      ]
+    })
+  }
+
+  managed_policy_arns = ["arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"]
+}
+
+
+resource "aws_iam_instance_profile" "support_sphere_instance_profile" {
+  name = "support-sphere-instance-profile"
+  role = aws_iam_role.support_sphere_instance_role.name
+}
+
+// security group
+resource "aws_security_group" "support_sphere_security_group" {
+  name = "support-sphere-security-group"
+  vpc_id = module.vpc.vpc_id
+
+  ingress {
+    from_port = 22
+    to_port = 22
+    protocol = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    from_port = 443
+    to_port = 443
+    protocol = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    from_port = 80
+    to_port = 80
+    protocol = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+// launch template -- userdata tbd for now
+
+resource "aws_launch_template" "support_sphere_launch_template" {
+  name = "support-sphere-launch-template"
+  // https://documentation.ubuntu.com/aws/en/latest/aws-how-to/instances/find-ubuntu-images/#finding-images-with-ssm
+  image_id = "resolve:ssm:/aws/service/canonical/ubuntu/server/jammy/stable/current/amd64/hvm/ebs-gp2/ami-id"
+  instance_type = "r5.large"
+  key_name = ""
+  #vpc_security_group_ids = [aws_security_group.support_sphere_security_group.id]
+  iam_instance_profile {
+    name = aws_iam_instance_profile.support_sphere_instance_profile.name
+  }
+
+  network_interfaces {
+    associate_public_ip_address = true
+    security_groups = [aws_security_group.support_sphere_security_group.id]
+  }
+
+  user_data = ""
+
+  update_default_version = true
+}
+
+// asg
+
+resource "aws_autoscaling_group" "support_sphere_asg" {
+  name = "support-sphere-asg"
+  launch_template {
+    id = aws_launch_template.support_sphere_launch_template.id
+    version = "$Latest"
+  }
+  min_size = 0
+  max_size = 1
+  desired_capacity = 0
+  vpc_zone_identifier = [module.vpc.public_subnets[0]]
+
+  dynamic "tag" {
+    for_each = data.aws_default_tags.default_tags.tags
+    content {
+      key = tag.key
+      value = tag.value
+      propagate_at_launch = true
+    }
+  }
+
+  lifecycle {
+    ignore_changes = [desired_capacity]
+  }
+}
+
+// Autoscaling action to shutdown the server every weekday at 1AM UTC (6PM PDT/5PM PST)
+// Replaces an overcomplicated lambda function/eventbridge rule setup
+
+resource "aws_autoscaling_schedule" "support_sphere_asg_schedule" {
+  scheduled_action_name = "support-sphere-asg-shutdown-after-working-hours"
+  min_size = 0
+  desired_capacity = 0
+  max_size = 1
+  recurrence = "0 1 * * MON-FRI"
```

Comment:
Question: Does this revert to the original autoscaling setting during work hours?

Reply:
no -- this just changes the desired capacity from whatever it was to 0

right now, my idea is that engineers will need to manually get the server started in order for the server to work, and that it won't just run by default

i can definitely change that if that's a less-than-ideal dev experience -- i'd just add another rule to set desired capacity to 1 at 8am pst/9am pdt

Reply:
Sounds good. This is fine and probably more money saving this way 😄